From 593d421b5b3666248454f5803e60712e0fbf5513 Mon Sep 17 00:00:00 2001
Message-Id: <593d421b5b3666248454f5803e60712e0fbf5513.1417780177.git.chang-joon.lee@intel.com>
In-Reply-To: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
References: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Wed, 29 Oct 2014 08:59:28 +0530
Subject: [PATCH 21/30] SQUASHME! [VPG]: drm/i915: Removed the extra backlight
 parsing from VBT

SQUASHME! - This patch should be squashed into the following existing patch:

Author: vkorjani <vikas.korjani@intel.com>
Date:   Fri Aug 1 05:33:41 2014 +0530
Change-Id: I09639dca0a5fcf1e684af78e70caf8528e76c621
	FOR_UPSTREAM [VPG]: drm/i915: Reading the PWM Frequency of BYT_CR from BIOS

Issue: GMINL-3283
Change-Id: Iff698db220dd596775156e4b1ce941aa00558c0f
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h    |    2 --
 drivers/gpu/drm/i915/intel_bios.c  |   18 ------------------
 drivers/gpu/drm/i915/intel_bios.h  |   12 ------------
 drivers/gpu/drm/i915/intel_panel.c |    2 +-
 4 files changed, 1 insertion(+), 33 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index a06f128..a199f2e 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1388,8 +1388,6 @@ struct intel_vbt_data {
 	} dsi;
 
 	int crt_ddc_pin;
-	/* Assuming only 1 LFP */
-	u16 pwm_frequency;
 
 	int child_dev_num;
 	union child_device_config *child_dev;
diff --git a/drivers/gpu/drm/i915/intel_bios.c b/drivers/gpu/drm/i915/intel_bios.c
index 493fb00..924de46 100644
--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -215,23 +215,6 @@ get_lvds_fp_timing(const struct bdb_header *bdb,
 	return (const struct lvds_fp_timing *)((const u8 *)bdb + ofs);
 }
 
-static void parse_backlight_data(struct drm_i915_private *dev_priv,
-		struct bdb_header *bdb)
-{
-	struct bdb_panel_backlight *vbt_panel_bl = NULL;
-	char *bl_start = NULL;
-
-	bl_start = find_section(bdb, BDB_LVDS_BACKLIGHT);
-	if (!bl_start) {
-		DRM_DEBUG_KMS("No backlight BDB found");
-		return;
-	}
-	DRM_DEBUG_KMS("Found backlight BDB");
-	vbt_panel_bl = (struct bdb_panel_backlight *)(bl_start + 1) +
-								panel_type;
-	dev_priv->vbt.pwm_frequency = vbt_panel_bl->pwm_freq;
-}
-
 /* Try to find integrated panel data */
 static void
 parse_lfp_panel_data(struct drm_i915_private *dev_priv,
@@ -1358,7 +1341,6 @@ intel_parse_bios(struct drm_device *dev)
 	parse_device_mapping(dev_priv, bdb);
 	parse_driver_features(dev_priv, bdb);
 	parse_edp(dev_priv, bdb);
-	parse_backlight_data(dev_priv, bdb);
 	parse_mipi(dev_priv, bdb);
 	parse_ddi_ports(dev_priv, bdb);
 
diff --git a/drivers/gpu/drm/i915/intel_bios.h b/drivers/gpu/drm/i915/intel_bios.h
index ebe8d01..ffca100 100644
--- a/drivers/gpu/drm/i915/intel_bios.h
+++ b/drivers/gpu/drm/i915/intel_bios.h
@@ -311,18 +311,6 @@ struct bdb_lvds_options {
 	u32 blt_control_type_bits;
 } __packed;
 
-struct bdb_panel_backlight {
-	/* Backlight control parameters */
-	u8 type:2;
-	u8 inverter_pol:1;
-	u8 gpio:3;
-	u8 gmbus:2;
-	u16 pwm_freq;
-	u8 minbrightness;
-	u8 i2c_slave_addr;
-	u8 brightnesscmd;
-} __packed;
-
 /* LFP pointer table contains entries to the struct below */
 struct bdb_lvds_lfp_data_ptr {
 	u16 fp_timing_offset; /* offsets are from start of bdb */
diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 6f20a42..0e92f9e 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -898,7 +898,7 @@ static void lpio_enable_backlight(struct drm_device *dev)
 	/* PWM enable
 	 * Assuming only 1 LFP
 	 */
-	pwm_base = compute_pwm_base(dev_priv->vbt.pwm_frequency);
+	pwm_base = compute_pwm_base(dev_priv->vbt.backlight.pwm_freq_hz);
 	pwm_base = pwm_base << 8;
 	lpio_bl_write(0, LPIO_PWM_CTRL, pwm_base);
 	lpio_bl_update(0, LPIO_PWM_CTRL);
-- 
1.7.9.5

